oci/private/pull.bzl

```diff
@@ -493,22 +494,25 @@ def _oci_alias_impl(rctx):
 
     downloader = _create_downloader(rctx)
 
+    available_platforms = []
+
+    manifest, _ = downloader.download_manifest(rctx.attr.identifier, "mf.json")
+
+    if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+        for submanifest in manifest["manifests"]:
+            parts = [submanifest["platform"]["os"], submanifest["platform"]["architecture"]]
+            if "variant" in submanifest["platform"]:
+                parts.append(submanifest["platform"]["variant"])
+            available_platforms.append('"{}"'.format("/".join(parts)))
+
     if _is_tag(rctx.attr.identifier) and rctx.attr.reproducible:
-        manifest, _ = downloader.download_manifest(rctx.attr.identifier, "mf.json")
+        is_bzlmod = hasattr(rctx.attr, "bzlmod_repository") and rctx.attr.bzlmod_repository
         digest = util.sha256(rctx, "mf.json")
-
         optional_platforms = ""
 
-        if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
-            platforms = []
-            for submanifest in manifest["manifests"]:
-                parts = [submanifest["platform"]["os"], submanifest["platform"]["architecture"]]
-                if "variant" in submanifest["platform"]:
-                    parts.append(submanifest["platform"]["variant"])
-                platforms.append('"{}"'.format("/".join(parts)))
-            optional_platforms = "'add platforms {}'".format(" ".join(platforms))
+        if len(available_platforms):
+            optional_platforms = "'add platforms {}'".format(" ".join(available_platforms))
 
-        is_bzlmod = hasattr(rctx.attr, "bzlmod_repository") and rctx.attr.bzlmod_repository
```

Comment:
how come we only set this variable conditionally on `_is_tag and reproducible` now, when we did it always before?

Reply:
no, we didn't do it always before. it was only set when the non-reproducible warning was printed which would have been printed only when `_is_tag and reproducible`

Reply:
now we fetch the manifest to know which platforms are available so that we can put it into build file.